wal/src/virtual_dom/vnode.rs

```diff
@@ -2,16 +2,21 @@ use web_sys::Node;
 
 use super::{VComponent, VElement, VList, VText};
 
+/// VNode is enum representing node in virtual DOM tree. Provides a wrapper over different types of nodes and concise and convinient API for vdom manipulation.
 #[derive(PartialEq, Debug)]
 pub enum VNode {
+    /// Represents [Element](https://developer.mozilla.org/en-US/docs/Web/API/Element) in DOM and contains [VElement],
     Element(VElement),
+    /// Represents [Text](https://developer.mozilla.org/en-US/docs/Web/API/Text) in DOM contains [VText],
     Text(VText),
+    /// Does not have direct representation in DOM tree, it represents custom component node and contains [VComponent],
```

Comment:
nie rozumiem tych komentarzy o VList i VComponent jakto one się zawierają

Reply:
One są na odwrót wgl. Bardziej niż to że component nie ma bezpośredniej reprezentacji, powiedział bym tu że on reprezentuje definowany przez użytkownika komponent